src/Simulation/TargetDefinitions/Decompositions/ApplyUncontrolledRy.qs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+
+    /// # Summary
+    /// Applies a rotation about the $y$-axis by a given angle. Note that the Controlled
+    /// functor is not supported.
+    ///
+    /// # Description
+    /// \begin{align}
+    ///     R_y(\theta) \mathrel{:=}
+    ///     e^{-i \theta \sigma_y / 2} = 
+    ///     \begin{bmatrix}
+    ///         \cos \frac{\theta}{2} & -\sin \frac{\theta}{2}  \\\\
+    ///         \sin \frac{\theta}{2} & \cos \frac{\theta}{2}
+    ///     \end{bmatrix}.
+    /// \end{align}
+    ///
+    /// # Input
+    /// ## theta
+    /// Angle about which the qubit is to be rotated.
+    /// ## qubit
+    /// Qubit to which the gate should be applied.
+    ///
+    /// # Remarks
+    /// Equivalent to:
+    /// ```qsharp
+    /// R(PauliY, theta, qubit);
+    /// ```
+    internal operation ApplyUncontrolledRy (theta : Double, qubit : Qubit) : Unit {
+        ApplyUncontrolledH(qubit);
+        ApplyUncontrolledS(qubit);
+        ApplyUncontrolledH(qubit);
+
+        ApplyUncontrolledRz(theta, qubit);
+
+        ApplyUncontrolledH(qubit);
+        ApplyUncontrolledSAdj(qubit);
+        ApplyUncontrolledH(qubit);
```

Comment:
Just curiosity:
Is there a way in Q# to get these subroutines called automatically in the same order?
(something like a mechanism of destructor calls in C++)

Reply:
Usually this pattern would make use of within-apply syntax (see [Conjugations in Q#](https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/statements/conjugations)), but that requires that the called operations support the  Adjoint functor, which these purposefully do not.